importer/main.py

```diff
@@ -396,6 +396,77 @@ def care_team_extras(
     return payload_string
 
 
+# This function takes in a payload_obj, a position and a tracker
+# It deletes the object at the given position in the resource->characteristic
+# It then returns the updated payload_obj and a new tracker for the remaining objects
+def delete_missing_obj_from_payload(payload_obj, position, tracker):
+    del payload_obj["resource"]["characteristic"][position]
+    return payload_obj, tracker-1
+
+
+# custom extras for product import
+def product_extras(resource, payload_string):
+    payload_obj = json.loads(payload_string)
+    product_name = resource[0]
+    tracker = 0
+
+    try:
+        (_, active, *_, previous_id, is_attractive_item, availability, condition, appropriate_usage,
+         accountability_period, image_source_url) = resource
+    except ValueError:
+        logging.error("Skipping: " + product_name + " : Because of missing columns ")
+        active = previous_id = is_attractive_item = availability = condition = appropriate_usage = \
+            accountability_period = image_source_url = "missing_column"
+
+    if active and active != "missing_column":
+        payload_obj["resource"]["active"] = active
+    else:
+        del payload_obj["resource"]["active"]
+
+    if previous_id and previous_id != "missing_column":
+        payload_obj["resource"]["identifier"][1]["value"] = previous_id
+    else:
+        del payload_obj["resource"]["identifier"][1]
+
+    if is_attractive_item and is_attractive_item != "missing_column":
```

Comment:
still so much repetition in this function, not sure it can be reduced without adding too much complexity?
open to suggestions

Reply:
one thing is to write a function `is_missing(col_name)` that returns a boolean, that'd simplify the conditionals

Reply:
actually see comment above for how to simplify

Reply:
ok so the reason for this whole tracker complexity is because you're calling del? Can you do this without calling del to avoid that? We need to find a way to remove the n+ with explicit numbers